Model/Api/Data/TaxData.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Model\Api\Data;
+
+use Bolt\Boltpay\Api\Data\TaxDataInterface;
+use Bolt\Boltpay\Api\Data\TaxResultInterface;
+use Bolt\Boltpay\Api\Data\ShippingOptionInterface;
+
+/**
+ * Class TaxData. Tax result and Shipping option properties of Tax.
+ *
+ * @package Bolt\Boltpay\Model\Api\Data
+ */
+class TaxData implements TaxDataInterface, \JsonSerializable
+{
+    /**
+     * @var TaxResultInterface
+     */
+    private $taxResult;
+
+    /**
+     * @var ShippingOptionInterface
+     */
+    private $shippingOption;
```

Comment:
Probably just revealing my own ignorance of M2's workings but if the shipping option has is doing it's own tax calculations, why does the tax data here need it?

Reply:
@njfaries The first call, Shipping, returns the available shipping options and their prices but without the tax calculated. The Tax endpoint applies the selected shipping option to cart and then calculates all the taxes and returns subtotal tax and tax on shipping separately. It's the result model as defined in docs and expected on Bolt end. The shipping option has the same structure in both Shipping (output) and Tax (input and output), but the tax on shipping data is populated in Tax and left empty (zero) in Shipping.